cmd/agent/daemon/app/app.go

```diff
@@ -256,6 +261,48 @@ func (a *App) Run(ctx context.Context) error {
 		return fmt.Errorf("proc handler: %w", err)
 	}
 
+	if cfg.EBPFMetrics.ProgramMetricsEnabled {
```

Comment:
Maybe this whole code block can be moved to some separate function.
Also goto is usually not a a good pattern for Go. Just moving to separate function would allow to use simple `return`

Reply:
Good point, I am going to switch over to returning the "return cleanup function" pattern.